pkgs/applications/virtualization/docker/default.nix

```diff
@@ -183,7 +196,7 @@ rec {
     '';
 
     preFixup = ''
-      find $out -type f -exec remove-references-to -t ${go} -t ${stdenv.cc.cc} '{}' +
+      find $out -type f -exec remove-references-to -t ${go.nativeDrv or go} -t ${stdenv.cc.cc} '{}' +
```

Comment:
In the case of cross-compilation, `go` refers to the host derivation, but we need the build derivation. `.nativeDrv` is only present when we're cross-compiling (this is the result of an optimization in `splice.nix`).

There are many cases similar in nixpkgs where a native dependency is meant to be interpolated into a string. For example, grep for `remove-references-to -t ${go}`. As in the case above, this does not work when cross-compiling.

There are no explicit references to `.nativeDrv` like this in nixpkgs. Such explicit references don't feel like the right solution here, especially considering the fact that the `.nativeDrv` attribute is unstable, and will someday be renamed to `pkgBuildHost`. Should we add library functions to wrap explicit access?

Reply:
Related: https://github.com/NixOS/nixpkgs/issues/49526#issuecomment-434806509 and https://github.com/NixOS/nixpkgs/pull/68895

Reply:
I think ```buildPackages.go``` would be better here? /cc @ericson2314

Reply:
Especially for removing references, the `go` interpolated into that string must match the one used as a `nativeBuildInput`. So, `buildPackages.go` would have to replace each instance of `go` in that file. The drawback of this approach is that `go` becomes harder to override for this package.

Reply:
Yes that is a the drawback, but splicing is evil in using things like `builtins.tryEval` and still falling over itself when the packages are too different. It's better to use `buildInputs.go`, I'm afraid. If you really want, you could do `goForBuild = buildPackages.go` in the override list or something and use that.

Reply:
I agree that this is the best solution in sight right now.